protocol/syncer.go

```diff
@@ -242,10 +242,11 @@ func (pw *progressionWrapper) runUpdateLoop(subscription blockchain.Subscription
 
 // stopProgression stops the progression tracking
 func (pw *progressionWrapper) stopProgression() {
+	pw.stopCh <- struct{}{}
```

Comment:
`stopProgression` shouldn't be holding a lock at the moment it's sending a stop signal to the stop channel the local runner is listening on - since it can cause a deadlock situation where the thread is holding a lock and waiting for someone to read from the `stopCh` and some other thread wants to grab the lock

Reply:
got it! thanks for the explanation 👍